eclass/cargo.eclass

```diff
@@ -195,12 +246,46 @@ cargo_gen_config() {
 	[term]
 	verbose = true
 	$([[ "${NOCOLOR}" = true || "${NOCOLOR}" = yes ]] && echo "color = 'never'")
+	$(_cargo_gen_git_config)
 	_EOF_
 
 	export CARGO_HOME="${ECARGO_HOME}"
 	_CARGO_GEN_CONFIG_HAS_RUN=1
 }
 
+# @FUNCTION: _cargo_gen_git_config
+# @USAGE:
+# @INTERNAL
+# @DESCRIPTION:
+# Generate the cargo config for git crates, this will output the
+# configuration for cargo to override the cargo config so the local git crates
+# specified in GIT_CRATES will be used rather than attempting to fetch
+# from git.
+#
+# Called by cargo_gen_config when generating the config.
```

Comment:
can you please add a` @CODE` example here? one simple case, one with subdirs like ruff.

Reply:
Done, I added a slightly simplified version the example from rustup, and one from ruff.